delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala

```diff
@@ -422,40 +422,57 @@ object Resources {
+      stateWhereResourceIsEditable(u).flatMap { s =>
+        val changesDetected = s.remoteContexts != remoteContextRefs || s.source != u.source
+        if(changesDetected) {
+          val schemaRef = u.schemaOpt.getOrElse(ResourceRef.Latest(s.schema.iri))
+          for {
+            (schemaRev, schemaProject) <- validate(u.id, expanded, schemaRef, s.project, u.caller)
+            time <- clock.realTimeInstant
+          } yield ResourceUpdated(u.id, u.project, schemaRev, schemaProject, types, u.source, compacted, expanded, remoteContextRefs, s.rev + 1, time, u.subject, u.tag)
+        } else {
+          // If no changes have been detected on the resource itself, we fallback on other commands or raise an error
+          (u.schemaOpt, u.tag) match {
+            case (Some(schema), tagOpt) => updateResourceSchema(UpdateResourceSchema(u.id, u.project, schema, u.rev, u.caller, tagOpt))
+            case (None, Some(tagValue)) => tag(TagResource(u.id, u.project, u.schemaOpt, s.rev, tagValue, u.rev, u.subject))
+            case (None, None) => IO.raiseError(NoChangeDetected(s))
+          }
+        }
+      }
       // format: on
     }
 
     def updateResourceSchema(u: UpdateResourceSchema) = {
       for {
-        s                          <- stateWhereResourceIsEditable(u)
-        (schemaRev, schemaProject) <- validate(u.id, u.expanded, u.schemaRef, s.project, u.caller)
-        types                       = u.expanded.getTypes.getOrElse(Set.empty)
+        s <- stateWhereResourceIsEditable(u)
+
+        (schemaRev, schemaProject) <- validate(u.id, s.expanded, u.schemaRef, s.project, u.caller)
+        _                          <- IO.raiseWhen(s.schema.iri == u.schemaRef.iri)(NoChangeDetected(s))
+        types                       = s.expanded.getTypes.getOrElse(Set.empty)
         time                       <- clock.realTimeInstant
-      } yield ResourceSchemaUpdated(u.id, u.project, schemaRev, schemaProject, types, s.rev + 1, time, u.subject)
+      } yield ResourceSchemaUpdated(u.id, u.project, schemaRev, schemaProject, types, s.rev + 1, time, u.subject, u.tag)
     }
 
-    def refresh(c: RefreshResource) = {
-      import c.jsonld._
+    def refresh(r: RefreshResource) = {
+      import r.jsonld._
       // format: off
       for {
-        s                          <- stateWhereResourceIsEditable(c)
-        _                          <- raiseWhenDifferentSchema(c, s)
-        (schemaRev, schemaProject) <- validate(c.id, expanded, c.schemaOpt.getOrElse(s.schema), s.project, c.caller)
+        s                          <- stateWhereResourceIsEditable(r)
+        _                          <- raiseWhenDifferentSchema(r, s)
+        (schemaRev, schemaProject) <- validate(r.id, expanded, r.schemaOpt.getOrElse(s.schema), s.project, r.caller)
+        noChangeDetected           = s.remoteContexts == remoteContextRefs && s.expanded == expanded
```

Comment:
As above, if the user wants to refresh but nothing changed, then validation doesn't matter?